internal/test/data/vector/gen_test.go

```diff
@@ -0,0 +1,665 @@
+package vector
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/test/goleak"
+)
+
+func TestFloat32VectorGenerator(t *testing.T) {
+	type args struct {
+		d   Distribution
+		n   int
+		dim int
+	}
+	type want struct {
+		n   int
+		dim int
+		err error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(args, want, Float32VectorGeneratorFunc, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(a args, w want, got Float32VectorGeneratorFunc, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if got != nil {
+			vectors := got(a.n, a.dim)
+			if len(vectors) != w.n && len(vectors[0]) != w.dim {
+				return errors.Errorf("got: \"%d\",\"%d\"\n\t\t\t\twant: \"%d\",\"%d\"", len(vectors), len(vectors[0]), w.n, w.dim)
+			}
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "gaussian distributed random vector test",
+			args: args{
+				d:   Gaussian,
+				n:   20,
+				dim: 10,
+			},
+			want: want{
+				n:   20,
+				dim: 10,
+				err: nil,
+			},
+		},
+		{
+			name: "uniform distributed random vector test",
+			args: args{
+				d:   Uniform,
+				n:   20,
+				dim: 10,
+			},
+			want: want{
+				n:   20,
+				dim: 10,
+				err: nil,
+			},
+		},
+		{
+			name: "unknown distribution error",
+			args: args{
+				d: -1,
+			},
+			want: want{
+				err: ErrUnknownDistribution,
+			},
+		},
+	}
```

Comment:
Almost LGTM
could you check the coding guile line?
https://vald.vdaas.org/docs/contributing/coding-style/#create-a-table-driven-test

Reply:
Sorry, I fixed the test case name.